### PR TITLE
[fix bug]issue #804 can not generate new block when diff increase,the…

### PR DESCRIPTION
**Pull request by frankcrypto2023**
#804

### CODE REVIEW

File: services/miner/cpuworker.go [hunk 1]
```diff
@@ -431,6 +432,8 @@ func (w *CPUWorker) solveBlock() *types.Block {
 		default:
 			// Non-blocking select to fall through
 		}
+		rand.New(rand.NewSource(time.Now().UnixNano()))
+		i := rand.Uint64()
```
Comment (lochjin) [inline services/miner/cpuworker.go]: instance.SetNonce(rand.Uint64())

**Comment by dindinw** [comment]
replaced by #809

**Comment by Devloper0007** [comment]
@dindinw @lochjin @frankcrypto2023 What was the reason for getting stuck on a high difficulty? Was it because the particular miner did not have enough computational power to calculate the nonce based on the previous logic?

I noticed that in the updated logic, a new random number is used for every attempt to generate the nonce, whereas previously, the process would go up to the max nonce value.

**Comment by dindinw** [comment]
@AyushSingh8x  
not, This issue is specially for the CPU miner, Especially for private network, no transaction, 2-node scenarios. The reason for this bug is that the difficulty adjustment algorithm makes the difficulty exceed the node's ability, making it impossible to calculate the answer within the specified time. Because there are no transactions and the block template itself is not updated, the node never has the opportunity to complete this task within the specified time on the next try. Using random nonces can alleviate this problem, but is not the source of the problem, so we use the #810 instead.

**Comment by Devloper0007** [comment]
ok @dindinw i will try this solution #810 and will update you on this

**Pull request closed without merging**